src/devices/Card/Ultralight/UltralightCard.cs

```diff
@@ -0,0 +1,992 @@
+﻿using System;
+using Iot.Device.Common;
+using Iot.Device.Ndef;
+using Microsoft.Extensions.Logging;
+
+namespace Iot.Device.Card.Ultralight
+{
+    /// <summary>
+    /// A Ultralight card class
+    /// </summary>
+    public class UltralightCard
+    {
+        /// <summary>
+        /// Default password used for write and read
+        /// </summary>
+        public static readonly byte[] DefaultPassword = new byte[4] { 0xFF, 0xFF, 0xFF, 0xFF };
+
+        // This is the actual RFID reader
+        private readonly CardTransceiver _rfid;
+        private readonly ILogger _logger;
+        private byte[]? _pack;
+
+        private byte _endAddress;
+
+        /// <summary>
+        /// The tag number detected by the reader, only 1 or 2
+        /// </summary>
+        public byte Target { get; set; }
+
+        /// <summary>
+        /// The NDEF capacity in bytes
+        /// </summary>
+        public int NdefCapacity { get; internal set; }
+
+        /// <summary>
+        /// The type of card
+        /// </summary>
+        public UltralightCardType UltralightCardType { get; internal set; }
+
+        /// <summary>
+        /// The block number to authenticate or read or write
+        /// </summary>
+        public byte BlockNumber { get; set; }
+
+        /// <summary>
+        /// The Data which has been read or to write for the specific block
+        /// </summary>
+        public byte[]? Data { get; set; }
```

Comment:
This looks like it should actually be a parameter to all the read/write functions, and clearly not a public writable property.

Reply:
It's unfortunately a bit more complicated than that. I used the same mechanism for the Mifare card. Yes, there are ways to make it more transparent but then it won't allow specific scenarios. For example, the way I did it for Mifare allowed user to have a minimum support for Ultralight (at least read and write). If I would have been doing that by hiding this part, it would not have been possible.
The higher level part, with NDEF for example totally hide all this complexity and this low level exposure.

Reply:
If I see it correctly, UltraLightCard is used directly by clients. You use it as high-level interface in your examples. Maybe you should add (in addition to the examples) a bit of documentation describing the overal structure and which high-level interfaces should be used for what. As a newbie, I would be quite confused as to where to start.

Reply:
Are the 3 samples not covering enough elements from the 3 readers?
I agree that NFC card are confusing. They really are. Most people are using it to get the ID and NDEF.